docs/content/4.sdk/3.advanced/3.custom-modules.md

```diff
@@ -0,0 +1,176 @@
+    },
+  } satisfies Connector;
+};
+```
+
+Now, the logger connector is in a separate file and can be used in the module.
+
+```typescript [module.ts]
+import { Module } from "@vue-storefront/sdk";
+import { loggerConnector } from "./connector";
+
+const loggerModule = () => {
+  return {
+    connector: loggerConnector(),
+    utils: {},
+    subscribers: {},
+  } satisfies Module;
+};
+```
+
+## Methods
+
+To keep the connector clean, you can separate the methods into a separate file. Create a new file `methods.ts` and add the following code:
+
+```typescript [methods.ts]
+export const log = (message: string) => {
+  console.log(message);
+};
+```
+
+Now, you can use the `log` method in the connector.
+
+```typescript [connector.ts]
+import { Connector } from "@vue-storefront/sdk";
+import * as methods from "./methods";
+
+const loggerConnector = () => {
+  return {
+    ...methods,
+  } satisfies Connector;
+};
+```
+
+Let's also make the methods more complex by extending their functionality. Let's assume that we want to log the message to the server as well.
+
+```typescript [methods.ts]
+export const log = (message: string) => {
+  console.log(message);
+  fetch("https://example.com/log", {
+    method: "POST",
+    body: JSON.stringify({ message }),
+  });
+};
+```
+
+Now, the `log` method is logging the message to the console and sending it to the server.
+
+## Utils
+
+Utils are the helper functions that can be helpful when working with the module. For example, you can create a helper function that is adding a timestamp.
+
+```typescript [module.ts]
+import { Module } from "@vue-storefront/sdk";
+import { loggerConnector } from "./connector";
+
+const loggerModule = () => {
+  return {
+    connector: loggerConnector(),
+    utils: {
+      formatMessage: (message: string) => {
+        return `${new Date().toISOString()} - ${message}`;
+      },
+    },
+    subscribers: {},
+  } satisfies Module;
+};
+```
+
+## Subscribers
+
+Subscribers are the functions that are called when the event is emitted. For example, you can create a subscriber that is logging the message to the server.
+
+```typescript [module.ts]
+import { Module } from "@vue-storefront/sdk";
+import { loggerConnector } from "./connector";
+
+const loggerModule = () => {
+  return {
+    connector: loggerConnector(),
+    utils: {
+      formatMessage: (message: string) => {
+        return `${new Date().toISOString()} - ${message}`;
+      },
+    },
+    subscribers: {
+      "*_after": (payload) => {
+        fetch("https://example.com/log", {
+          method: "POST",
+          body: JSON.stringify(payload),
+        });
+      },
+    },
+  } satisfies Module;
+};
+```
+
+Now, the subscriber is logging the payload to the server after every method call of every SDK module.
+
+Finally, you can use the module in your application.
+
+```typescript
+import { initSdk, buildModule } from "@vue-storefront/sdk";
+import { loggerModule } from "./module";
```

Comment:
Similar to the comment about exports. None of our code snippets export this module.

Reply:
Fixing, ty